src/sentry/spans/consumers/process/factory.py

```diff
@@ -119,25 +148,31 @@ def _produce_segment(message: Message[ProduceSegmentContext | None]):
             if len(keys) > 0:
                 payload_context["sample_key"] = keys[0]
 
-            total_spans_read = 0
             # With pipelining, redis server is forced to queue replies using
             # up memory, so batching the keys we fetch.
             with txn.start_child(op="process", description="produce_fetched_segments"):
                 for i in range(0, len(keys), BATCH_SIZE):
                     segments = client.read_and_expire_many_segments(keys[i : i + BATCH_SIZE])
 
                     for segment in segments:
-                        total_spans_read += len(segment)
-                        produce_segment_to_kafka(segment)
+                        if not len(segment):
+                            continue
 
-            sentry_sdk.set_context("payload", payload_context)
+                        payload_data = prepare_buffered_segment_payload(segment)
+                        buffered_segments.append(KafkaPayload(None, payload_data, []))
 
+    if not len(buffered_segments):
```

Comment:
Technically, you can initialize this list with a FILTERED_PAYLOAD to make sure it'll always commit and just append, it'll be ignore on produce anyway. Save a `if`.